spark/unitycatalog/src/test/java/io/sparkuctest/UCDeltaTableIntegrationBaseTest.java

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.sparkuctest;
+
+import org.apache.spark.SparkConf;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Abstract base class for Unity Catalog + Delta Table integration tests.
+ *
+ * This class provides a pluggable SQL execution framework via the SQLExecutor interface,
+ * allowing tests to be written once and executed via different execution engines
+ * (e.g., Spark SQL, JDBC, REST API, etc.).
+ *
+ * Subclasses must provide an executor by implementing the getSqlExecutor method.
+ */
+public abstract class UCDeltaTableIntegrationBaseTest extends UnityCatalogSupport {
+
+  /**
+   * Provides all table types for parameterized tests.
+   * Tests can use this as a @MethodSource to test different table types.
+   */
+  protected static Stream<TableType> allTableTypes() {
+    return Stream.of(TableType.EXTERNAL, TableType.MANAGED);
+  }
+
+  private static SparkSession sparkSession;
+
+  /**
+   * Create the SparkSession before all tests.
+   */
+  @BeforeAll
+  public static void setUpSpark() throws Exception {
+    // UC server is started by UnityCatalogSupport.setup()
+    // And the BeforeAll of parent class UnityCatalogSupport will be called before this method.
```

Comment:
Thanks for the comment, it exactly explain when the unitycatalog server will start.